design_drafts/transmission_loading_and_parametrization.md

```diff
@@ -0,0 +1,194 @@
+    </actuator>
+    <actuator name="bar_actuator">
+      <role>actuator2</role>
+      <mechanicalReduction>-50</mechanicalReduction>
+    </actuator>
+  </transmission>
+
+</robot>
+```
+
+Let's dissect the XML above:
+
+* `<type>transmission_interface/FourBarLinkageTransmission</type>` defines the transmission plugin to be loaded, parameters are following
+* `<joint name="bar_joint">` defines the role and hardware interface of joints. The joint `bar_joint` should already be defined in the URDF.
+* `<role>joint2</role>` is internal to transmission implementations, we'll not explain that here
+* `<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>` defines the *joint interface type* which the transmission needs to know for picking the right formula to apply to the values, e.g. a reduction rate may have a different effect to position, velocity or effort values
+* the section enclosed by `<actuator name="foo_actuator">` defines the actuator name, role and mechanical reduction rate to be used in the underlying formula.
+
+Let's look at some issues now.
+
+### More than one role
+
+This is the first and only place *joint interface type* is defined for a joint. It should not be the responsibility of transmissions to carry this information but since the URDF has no support for it, it had to be added here.
+
+Throughout ROS1 there are many robots using `SimpleTransmission` with a reduction rate of 1 only to be able to define the type of joint interfaces they use so `gazebo_ros_control` can use the intended joint interfaces and simulate their robot properly.
+
+### Actuator names are not really defined
+
+Actuators are virtually non-existent in the URDF, the first and only place they are shown is in the transmission definition. It is entirely up to the one implementing the `RobotHW` to make sense of these names.
+
+Common policies are to match `bar_joint` to `bar_actuator` or `joint1` to `joint1_motor`.
+
+### Joint and actuator tags
+
+There are some minor issues that cause confusion from time to time:
+* Why is the `hardwareInterface` tag paired with the joint? Could be defined for both if we want to allow different interfaces or if we want to ensure that they are not different for compatibility...
+* Why is the `mechanicalReduction` tag paired with the actuator? The reduction is applied to one of them when the information flows from actuator to controller but the inverse is applied when it flows from controller to actuator.
+
+
+
+## Proposals for ROS2 XML format
+
+The plugin-based system should be kept for transmissions. The tag `type` was replaced with `plugin` in the xml format for clarity and consistency with the rest of `ros2_control`.
+
+The parametrization should become simpler as it doesn't need to concern itself anymore with the joint command interface types as they are defined elsewhere.
+
+It is however still unclear if there is any use to define actuator names outside the transmission scope.
```

Comment:
This is something I wonder too. In any case it would fall outside the scope of this task though.